samples/detect.v1p2beta1.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2018, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function detectPdfText(bucketName, fileName) {
+  // [START vision_async_detect_document_ocr]
+
+  // Imports the Google Cloud client libraries
+  const vision = require('@google-cloud/vision').v1p2beta1;
+
+  // Creates a client
+  const client = new vision.ImageAnnotatorClient();
+
+  const gcsSourceUri = `gs://${bucketName}/${fileName}`;
+  const gcsDestinationUri = `gs://${bucketName}/${fileName}.json`;
+
+  console.log(gcsDestinationUri);
+
+  const request = {
```

Comment:
Ouch.

This request is pretty **rough** to have to build. Seriously.

I don't have any idea how to solve this at this moment in time, so it's not necessarily a blocker. But... **OUCH.**

/cc fyi @lukesneeringer @jmdobry @ace-n

Reply:
Is the Ouch for the sample code, or for the way the API is designed?

Reply:
**tl;dr** it LGTM, no worries 

IMO the API has a reasonable design, Features is standard in Vision, and input/output configs seem clear to me and they follow GCS uri conventions, I believe.

The request object is really disgusting to construct, however.

23 LOC are *just* for constructing the request object in this sample. The request does not look idiomatic (to me). As a developer, I wonder why the library doesn't have some kind of Helper for GCS uris.

**API** OK
**Library** SAD
**Sample** OK, stuck with the library we have

I don't think it's worth blocking, but it might be worth a bug on https://github.com/toolkit/toolkit requesting a shorthand for constructing requests with GCS uris and Features? Meh. Not sure! The code just makes me a little queezy.

For EVERY OTHER Vision sample, the library provides helpers so you don't need to explicitly call the *batch* API when you're only running detection on 1 file. It's annoying that this doesn't follow the same convention and developers need to pass multiple request objects.

Reply:
Yeah, I agree with you. This API and library was not intuitive to understand.